UnityProject/Assets/Scripts/UI/Cargo/GUI_SecurityRecords.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class GUI_SecurityRecords : NetTab
+{
+	[SerializeField]
+	private NetPageSwitcher nestedSwitcher;
+	[SerializeField]
+	private GUI_SecurityRecordsEntriesPage entriesPage;
+	[SerializeField]
+	private GUI_SecurityRecordsEntryPage entryPage;
+	[SerializeField]
+	private NetLabel idText;
+	private SecurityRecordsConsole console;
+	public IDCard InsertedCard;
+
+	public override void OnEnable()
+	{
+		base.OnEnable();
+		StartCoroutine(WaitForProvider());
```

Comment:
is this coroutine necessary? I'm not sure of what you're trying to do

Reply:
Well, I've seen it in GUI_Paper, it is used so I update info only when there is a Provider
If I just put this into OnEnable - it will give NRE